src/assisted_test_infra/test_infra/controllers/assisted_installer_infra_controller.py

```diff
@@ -233,8 +248,10 @@ def wait_for_pod_change(self, name: str, replicas_count: int, timeout=180, sleep
             def pods_are_running():
                 # Waiter lambda verify all pods are running begins with the same name
                 pods = self._get_obj_resources_by_name(self.Resources.POD.value, name)
-                running = list(map(lambda pod: hasattr(pod.status, "phase") and pod.status.phase == "Running", pods))
-                return all(running)
+                running_pods = list(map(self.containers_ready_inside_pod, pods))
+                # make sure all pods are running , in restartAt we may have parallel creation/termination
+                logging.info(f"Check if total pods are equal to running and ready {running_pods} vs {len(pods)}")
+                return len(pods) == len(running_pods) and all(running_pods)
```

Comment:
could be simpler to check if the full deployment has been rolled out successfully?

You can get inspiration from here:
https://github.com/fiaas/fiaas-deploy-daemon/pull/47/files#diff-2a85b9b3dfebaeb782d011387395182908f17ba26d491c029850f6ddcae02cecR66-R69

it does the same check as `kubectl rollout status ...`